### PR TITLE
Use strdup and fix all casts 

### PR DESCRIPTION
**Pull request by metsma**
Signed-off-by: Raul Metsma <raul@metsma.ee>

##### Checklist

- [ ] Documentation is added or updated
- [ ] New files have a LGPL 2.1 license statement
- [X] PKCS#11 module is tested
- [ ] Windows minidriver is tested
- [ ] macOS tokend is tested

### CODE REVIEW

**Comment by frankmorgner** [comment]
Does the card have a valid EF.TokenInfo? If so, the synthetic profile parsing of EF.TokenInfo will allocate resources, which this PR will not free when the emulated profile is loaded.

**Comment by metsma** [comment]
> Does the card have a valid EF.TokenInfo? If so, the synthetic profile parsing of EF.TokenInfo will allocate resources, which this PR will not free when the emulated profile is loaded.

No

**Comment by martinpaljak** [comment]
I looked at the code dealing with the emulation parts and binding and think that life has changed from when that was written. options like turning emulation off and the order of detection and the number of emulated drivers... Maybe we should change that.

**Comment by frankmorgner** [comment]
What do you have in mind? SC_CARD_TYPE_MCRD_ESTEID_V30 is already listed in sc_pkcs15_is_emulation_only...

File: src/libopensc/pkcs15-esteid.c [hunk 1]
```diff
@@ -166,7 +150,7 @@ sc_pkcs15emu_esteid_init (sc_pkcs15_card_t * p15card)
 		pin_info.attrs.pin.stored_length = 12;
 		pin_info.attrs.pin.max_length = 12;
 		pin_info.attrs.pin.pad_char = '\0';
-		pin_info.tries_left = (int)tries_left;
+		pin_info.tries_left = buff[5];
```
Comment (frankmorgner) [inline src/libopensc/pkcs15-esteid.c]: you need to check the length

**Pull request merged**